pkg/logging/http.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package logging
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+
+	"net/http"
+	"time"
+	"github.com/thanos-io/thanos/pkg/server/http/middleware"
+)
+
+type LogServerMiddleware struct {
```

Comment:
```suggestion
type ServerMiddleware struct {
```
It's `logging` package after all (:

Reply:
Also, I'd love to see an interface for the Log Middleware

Reply:
Can you elaborate on the details of the interface? @kakkoyun

Reply:
More info - https://github.com/thanos-io/thanos/issues/2914